text/0043-block-metadata.md

```diff
@@ -0,0 +1,238 @@
+* Feature Name: `block_metadata`
+* Start Date: 2022-05-24
+* MCIP PR: [mobilecoinfoundation/mcips#43](https://github.com/mobilecoinfoundation/mcips/pull/43)
+* Tracking Issue: https://github.com/mobilecoinfoundation/mobilecoin/issues/2104
+
+Contents:
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Guide-level explanation](#guide-level-explanation)
+- [Reference-level explanation](#reference-level-explanation)
+  - [Protobuf schema](#protobuf-schema)
+  - [Signing metadata](#signing-metadata)
+  - [Signing key history](#signing-key-history)
+  - [Historical AVRs](#historical-avrs)
+  - [Block Version bump](#block-version-bump)
+- [Drawbacks](#drawbacks)
+- [Rationale and alternatives](#rationale-and-alternatives)
+  - [Alternatives considered](#alternatives-considered)
+- [Prior art](#prior-art)
+- [Unresolved questions](#unresolved-questions)
+- [Future possibilities](#future-possibilities)
+
+# Summary
+[summary]: #summary
+
+Add metadata to the blockchain, with consensus nodes reporting the Attested
+Verification Report and quorum set at the time a block was externalized.
+
+# Motivation
+[motivation]: #motivation
+
+This is a long-awaited feature for the MobileCoin blockchain.
+
+The AVR proves that the block was signed by an enclave, as the pubkey used to
+sign the block is in the AVR. Signing the metadata with the SCP message signing
+key proves that the block and metadata were generated by the holder of that
+private key.
```

Comment:
This is more of a summary than a motivating statement

```suggestion
Currently if one wants to prove blocks were generated by the MobileCoin network (say
instead of a forged chain) - the information required to prove this is hard to obtain.

In order to ensure chain integrity to participants in the mobilecoin network
and those auditing it. The following information is proposed to be distributed
on the blockchain itself:
1. An AVR that proves a block was signed by an SGX enclave signed by the Intel chain
of trust
2. Block Metadata including the node's quorum set, AVR, and SCP message signing key 
signed by the SCP message signing key to prove that the node was the holder of that
private key when the block was signed
```